compiler/z/codegen/OpMemToMem.cpp

```diff
@@ -1809,7 +1809,7 @@ MemToMemTypedVarLenMacroOp::generateLoop()
 
    // Skip the loop if length is zero.
    TR::LabelSymbol * doneLoop = TR::LabelSymbol::create(_cg->trHeapMemory(),_cg);
-   startControlFlow = generateS390CompareAndBranchInstruction(_cg, TR::InstOpCode::getCmpOpCodeFromNode(_lenNode), _dstNode, _lenReg, (int32_t)0, TR::InstOpCode::COND_BNH, doneLoop, false, false);
+   startControlFlow = generateS390CompareAndBranchInstruction(_cg, TR::InstOpCode::getCmpOpCode(), _dstNode, _lenReg, (int32_t)0, TR::InstOpCode::COND_BNH, doneLoop, false, false);
```

Comment:
The reason this and the subsequent change is correct is because the type of the `TR::arraycopy` node will be the same width as the system pointer type. This can clearly be seen here:

https://github.com/eclipse/omr/blob/e0b8377e50cbc004895cd3aa36e0c3f05dba5322/compiler/optimizer/ValuePropagationCommon.cpp#L871-L932

These `MemToMemTypedVarLenMacroOp` are only used in the context of evaluating `TR::arraycopy`, and hence `getCmpOpCodeFromNode` is equivalent to `getCmpOpCode`. Furthermore a few lines below the context in which `_lenReg` is used also reinforces the fact that on 64-bit target we need to use 64-bit compares and on 32-bit target we need to use 32-bit compares.